File: calebpagel+clojure/fizzbuzz.clj
;; Learning Clojure ....

;; 

;; Comments welcome


(def buzz (fn [x] (= (mod x 5) 0)))


(def fizz (fn [x] (= (mod x 3) 0)))


(def fizzbuzzer (fn [x] (if (and (fizz x) (buzz x)) "FIZZBUZZ"
                          				(if (fizz x) "FIZZ" 
								(if (buzz x) "BUZZ" 
                              						x)))))

(dotimes [n 100] (println n "is" (fizzbuzzer n)))

File: niftyn8+haskell/fizzbuzz.hs
fizzbuzz :: Int => String
fizzbuzz x
  | (mod x 15) == 0 = "FizzBuzz"
  | (mod x 3) == 0 = "Fizz"
  | (mod x 5) == 0 = "Buzz"
  | otherwise = show x

main :: IO()
main = mapM_ putStrLn (map fizzbuzz [1..10000])
